apps/librelingo_audios/librelingo_audios/generate_audios.py

```diff
@@ -0,0 +1,88 @@
+from librelingo_utils import audio_id, iterate_phrases
+from pathlib import Path
+import subprocess
+import json
+import random
+
+def generate_audios_for_course(output_path, course_name, course, settings):
+    if not course.settings.audio_settings.enabled:
+        return
+
+    index_path = Path(Path(output_path) / "{}.json".format(course_name))
+
+    phrases_in_course = [(_index_dict_for_phrase(p, course), p) for p in iterate_phrases(course)]
+
+    if settings.overwrite:
+        # Pretend nothing already exists; recreate everything
+        phrase_indices_already_present = []
+    else:
+        with open(index_path, "r") as f:
+            phrase_indices_already_present = json.loads(f.read())["audios"]
+
+    result_index = []
+
+    # For each phrase in the course, if it doesn't exist, generate it.
```

Comment:
What do you think about the idea of extracting this logic into a new function?

```python
result_index = _load_result_index()
```

It could basically combine this logic:

```python
        with open(index_path, "r") as f:
            phrase_indices_already_present = json.loads(f.read())["audios"]
```

with also the deduplication logic (at least I assume the logic below is for deduplication?)

Reply:
Good idea. I'm actually refactoring the whole method to make it clearer. The logic below was not for deduplication, so clearly it could use some refactoring :)